indexer/src/candidates_validator.rs

```diff
@@ -141,19 +155,21 @@ impl CandidatesValidator {
     async fn process_candidates(self) -> Result<()> {
         let Self {
             mut receiver,
-            rabbit_publisher,
+            rabbit_publisher_handle,
             view_client,
+            listener,
         } = self;
 
         let queue_protected = sync::Arc::new(Mutex::new(VecDeque::new()));
-        let publisher_protected = sync::Arc::new(Mutex::new(rabbit_publisher));
+        let publisher_protected = sync::Arc::new(Mutex::new(rabbit_publisher_handle));
 
         let (done_sender, done_receiver) = mpsc::channel(1);
```

Comment:
I've missed that, but look like you need the `oneshot` channel here.

Reply:
Done